src/hotspot/share/memory/metaspace.cpp

```diff
@@ -672,20 +672,23 @@ void Metaspace::global_initialize() {
 
   metaspace::ChunkHeaderPool::initialize();
 
+  if (DumpSharedSpaces) {
+    assert(!UseSharedSpaces, "sanity");
+    MetaspaceShared::initialize_for_static_dump();
+  }
+
   // If UseCompressedClassPointers=1, we have two cases:
-  // a) if CDS is active (either dump time or runtime), it will create the ccs
+  // a) if CDS archive has been mapped, it will create the ccs
```

Comment:
This comment is confusing. So, at this point the ccs archive has already be mapped? Where?

Reply:
I've changed it to:

```
  // a) if CDS archive(s) have been mapped, we would have already reserved the
  //    ccs (immediately above the mapped archives) and initialized
  //    the CompressedKlassPointers encoding.
```

Reply:
No, the confusing part is the order. The old comment used future tense, this uses past tense. So to the reader it looks like mapping may already have happened. But at this point in the code the archive has not been mapped yet, no?
Proposal:
```
  // If UseCompressedClassPointers=1, we have two cases:
  // a) if CDS is active (runtime, Xshare=on), it will create the class space
  //    for us, initialize it and set up CompressedKlassPointers encoding.
  //    Class space will be reserved above the mapped archives.
  // b) if CDS either deactivated (Xshare=off) or a static dump is to be done (Xshare:dump),
  //    we will create the class space on our own. It will be placed above the java heap,
  //    since we assume it has been placed in low
  //    address regions. We may rethink this (see JDK-8244943). Failing that,
  //    it will be placed anywhere.
```

Reply:
This looks good. I added it to the code.